src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75d;
+    private int threshold;
+    private int capacity;
+    private int size;
+    private Node<K, V>[] data;
+
+    public MyHashMap() {
+        capacity = DEFAULT_CAPACITY;
+        data = new Node[capacity];
+        threshold = (int) (capacity * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node currentNode;
+        Node newNode = new Node<>(key, value, null);
+        int index = getBucketByKey(key);
+        if (size > threshold) {
+            resize();
+        }
+        if (data[index] == null) {
+            data[index] = newNode;
+        } else {
+            currentNode = data[index];
+            while (currentNode.next != null) {
+                if (Objects.equals(currentNode.key, key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+                return;
+            }
+            currentNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node node = data[getBucketByKey(key)];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return (V) node.value;
```

Comment:
```suggestion
                return node.value;
```
this cast is redundant